packages/manifold/test/physicalclaim/PhysicalClaimBase.t.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "../../contracts/physicalclaim/IPhysicalClaim.sol";
+
+abstract contract PhysicalClaimBase is Test {
+
+    uint256 internal privateKey = 0x1010101010101010101010101010101010101010101010101010101010101010;
+
+    function constructSubmission(uint256 instanceId, IPhysicalClaim.BurnToken[] memory burnTokens, uint8 variation, uint64 variationLimit, uint64 totalLimit, address erc20, uint256 price, address payable fundsRecipient, uint160 expiration, bytes32 nonce) internal view returns (IPhysicalClaim.BurnSubmission memory submission) {
+        // Hack because we were getting stack too deep, so need to pass into subfunction
+        submission = _constructSubmission(instanceId, burnTokens, variation, variationLimit, totalLimit, erc20, price, fundsRecipient, expiration, nonce);
+        submission.instanceId = instanceId;
+    }
+
+    function _constructSubmission(uint256 instanceId, IPhysicalClaim.BurnToken[] memory burnTokens, uint8 variation, uint64 variationLimit, uint64 totalLimit, address erc20, uint256 price, address payable fundsRecipient, uint160 expiration, bytes32 nonce) internal view returns (IPhysicalClaim.BurnSubmission memory submission) {
+        bytes memory messageData = abi.encode(instanceId, burnTokens, variation, variationLimit, totalLimit, erc20, price, fundsRecipient, expiration, nonce);
+        bytes32 message = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", messageData));
```

Comment:
Is this string correct? I can't find exactly what that \n32 escape character does but I'm suspicious it's not just a newline followed by '32'

Reply:
pretty sure it's some sort of encoded space or new line.

Reply:
It is 100% correct.  This is how you decode ethereum signed messages and we use it in a number of places.

Reply:
Actually let me double check. But the 32 represents the size of data

Reply:
Doesn't the unit test pass?

Reply:
Hmmm, 32 might not be the right length here.

Reply:
Found it yes you are correct:
https://eips.ethereum.org/EIPS/eip-191

It's a new line followed by bytes.

> Prompting the user to sign the message with [signMessage()](https://docs.ethers.io/v5/api/signer/#Signer-signMessage) which prefixes the Ethereum ([EIP-191](https://eips.ethereum.org/EIPS/eip-191)) specific identifier "\x19Ethereum Signed Message:\n" and byte length of the message. This

Reply:
I mean, it 'works' but it's wrong.